File: abc_323/D/main.py
#!/usr/bin/env python3
N = int(input())
S = [0 for _ in range(N)]
C = [0 for _ in range(N)]
di = {}
for i in range(N):
    S[i], C[i] = map(int, input().split())
    di[S[i]] = C[i]

soS = sorted(S)
for i in range(N):
    size = soS[i]
    cnt = di[size]
    flag = True
    while cnt > 1:
        mod = cnt % 2
        cnt = cnt // 2
        di[size] = mod
        size = 2 * size
        if size in di:
            di[size] += cnt
            flag = False
            # print("#", di)
            break
        # print(di)
    if flag:
        di[size] = cnt
# for tmp in list(di.items()):
#     print(tmp)
print(sum(list(di.values())))

File: abc_323/D/main.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	var n int
	var s, c []int
	di := make(map[int]int)
	fmt.Scan(&n)
	for i := 0; i < n; i++ {
		var ts, tc int
		fmt.Scan(&ts, &tc)
		s = append(s, ts)
		c = append(c, tc)
		di[ts] = tc
	}
	sort.Slice(s, func(i, j int) bool { return s[i] < s[j] })
	ans := 0
	for _, num := range s {
		size := num
		cnt := di[size]
		for cnt > 0 {
			mod := cnt % 2
			cnt = cnt / 2
			ans += mod
			size *= 2
			if _, ok := di[size]; ok {
				di[size] += cnt
				break
			}
			if cnt == 1 {
				ans += 1
				break
			}
		}
	}
	fmt.Println(ans)
}
